src/components/popup/details/vendors/vendors.jsx

```diff
@@ -61,6 +65,14 @@ export default class Vendors extends Component {
 		isSelected ? this.handleAcceptAll() : this.handleRejectAll();
 	};
 
+	isFullLegitimateInterestChosen = () => {
+		const {vendorsWithLegIntsIds, selectedLegitimateInterestsIds} = this.props;
+		const isSelected = (el) => selectedLegitimateInterestsIds.has(el);
+		return vendorsWithLegIntsIds.every(isSelected);
+	};
+
+	handleFullLegIntChange = ({isSelected}) => this.props.selectAllVendorLegitimateInterests(isSelected);
```

Comment:
You shouldn't use selectAllVendorLegitimateInterests function here because it will impact every vendor which have at least 1 legIntPurpose. Instead of that, you should select/unselect vendors that are passed to this component. I.e: 

`handleFullLegIntChange  = () => {
		const {vendors, selectVendorLegitimateInterests} = this.props;
		selectVendorLegitimateInterests(vendors.map(({id}) => id), true)
};`

- [ ] selectVendorLegitimateInterests instead of selectAllVendorLegitimateInterests